android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/StorageCipher18Implementation.java

```diff
@@ -33,22 +34,26 @@ public StorageCipher18Implementation(Context context) throws Exception {
         SharedPreferences.Editor editor = preferences.edit();
 
         String aesKey = preferences.getString(AES_PREFERENCES_KEY, null);
-        byte[] key;
-
-        if (aesKey == null) {
-            key = new byte[keySize];
-            secureRandom.nextBytes(key);
-            secretKey = new SecretKeySpec(key, KEY_ALGORITHM);
-
-            byte[] encryptedKey = rsaCipher.wrap(secretKey);
-            editor.putString(AES_PREFERENCES_KEY, Base64.encodeToString(encryptedKey, Base64.DEFAULT));
-            editor.commit();
-        } else {
-            byte[] encrypted = Base64.decode(aesKey, Base64.DEFAULT);
-            secretKey = rsaCipher.unwrap(encrypted, KEY_ALGORITHM);
-        }
 
         cipher = Cipher.getInstance("AES/CBC/PKCS7Padding");
+
+        if (aesKey != null) {
+            try {
+                byte[] encrypted = Base64.decode(aesKey, Base64.DEFAULT);
+                secretKey = rsaCipher.unwrap(encrypted, KEY_ALGORITHM);
+                return;
+            } catch (Exception e) {
+                Log.e("StorageCipher18Impl", "unwrap key failed", e);
+            }
+        }
+
+        byte[] key = new byte[keySize];
+        secureRandom.nextBytes(key);
```

Comment:
Doesn’t it create new key every time ?
Also shouldn’t be data deleted too?

Reply:
No, I don't. You maybe get me wrong. If the current key can be unwrapped, I use this key and I call `return` so the codes of creating a new key are ignored.

Reply:
Yes, my fault, haven't noticed return :+1:

What's about the data which was encrypted with "broken" key?

Reply:
yep, I forgot to delete data of the wrong key